django/cantusdb_project/main_app/views/chant.py

```diff
@@ -1139,45 +1097,74 @@ def get_suggested_chants(self):
+            cantus_id: str,
+            count: int,
+        ) -> dict:
+            """
+            Request information from Cantus Index about a given Cantus ID,
+            adding keys "count" and "genre_id" to this dictionary and returning it.
+
+            For the following explanation, assume the user has just added a chant
+            with a Cantus ID of "123456" to the database
 
             Args:
-                suggested_chant (tuple(str, int)): tuple containing the cantus ID of a chant,
-                along with an int that represents the number of times the suggested chant follows
-                another particular chant.
+                cantus_id (str): a Cantus ID (i.e. we've looked through the database
+                    for chants with a Cantus ID of "123456", and we've found all the
+                    chants that follow those chants. `cantus_id` is the Cantus ID of
+                    one of those following chants.)
+                count (int): The number of times a chant with a Cantus ID of `cantus_id`
+                    follows a chant with a Cantus ID of "123456" among all sources in
+                    the database
 
             Returns:
-                dict: dictionary containing data for a specific chant, along with a count
-                of how many times it appears after instances of the other particular chant.
+                dict: a dictionary with data about a specific Cantus ID, including:
+                    - a canonical fulltext (str)
+                    - an incipit (str)
+                    - a genre (str)
+                    - the ID of that genre in Cantus Database (int)
+                    - how many times chants with this Cantus ID follow chants with a
+                        Cantus ID of 123456 (int)
             """
-            sugg_chant_cantus_id, sugg_chant_count = suggested_chant
-            # search Cantus Index
-            response = requests.get(
-                f"https://cantusindex.org/json-cid/{sugg_chant_cantus_id}",
-                timeout=5,
-            )
-            assert response.status_code == 200
-            # parse the json export to a dict
-            # can't use response.json() because of the BOM at the beginning of json export
-            chant_dict = json.loads(response.text[2:])[0]
-            # add number of occurence to the dict, so that we can display it easily
-            chant_dict["count"] = sugg_chant_count
-            # figure out the id of the genre of the chant, to easily populate the Genre selector
-            genre_name = chant_dict["genre"]
-            genre_id = Genre.objects.get(name=genre_name).id
-            chant_dict["genre_id"] = genre_id
-            return chant_dict
+            try:
+                response = requests.get(
+                    f"https://cantusindex.org/json-cid/{cantus_id}",
+                    timeout=5,
+                )
+                assert response.status_code == 200
```

Comment:
I'm not sure it's worth refactoring these and the other try/except wrappers, but I don't love the bundling of all these possible exceptions together...if you ever were trying to use these to figure out what happened: did the server not respond? Was the result mal-formed? You're also doing a database query of your own database within the try...

Reply:
I've refactored this, and split it up into a few different `except` cases - the difference is only in the error message printed to the console, but things should be a little bit clearer now.

Reply:
I find the `assert` to be weird. You might consider instead using the context manager version of the `requests` get... 

```
with requests.Session() as s:
    s.get(...url...)
```

https://requests.readthedocs.io/en/latest/user/advanced/#session-objects